src/OptFrame/Heuristics/SA/BasicSimulatedAnnealing.hpp

```diff
@@ -276,13 +281,14 @@ class BasicSimulatedAnnealingBuilder : public SA
    {
    }
 
+   // has sptr instead of sref, is that on purpose or legacy class?
```

Comment:
Interesting point. In this case, we want to have the ability to load "nullptr references", meaning that we want std::shared_ptr pure and simple. Then, once we convert into sref, it would naturally break if its null. If we want to give a better treatment to that, in the future, we have the option of returning a null SingleObjSearch<XES>*, representing an "optional reference", which is again another std::shared_ref with null. But not necessary for now.